core/src/ledger/parameters/storage.rs

```diff
@@ -13,17 +13,37 @@ const POS_GAIN_D_KEY: &str = "pos_gain_d";
 const STAKED_RATIO_KEY: &str = "staked_ratio_key";
 const POS_INFLATION_AMOUNT_KEY: &str = "pos_inflation_amount_key";
 
+// keep these keys in sync with the defs above;
+// make sure to store them in sorted order!
+const ALL_KEYS: &[&str] = &[
```

Comment:
maybe we could add a test case to check that they are indeed sorted

Reply:
yeah, we can't test if all the keys were included in the slice though :(

Reply:
but I can write a test to check if the keys are sorted!

Reply:
Yeah, destructuring a struct pattern is great for this. We can do something like:

```rust
struct Keys {
    epoch_duration: &'static str,
    vp_whitelist: &'static str,
    ...
}

const KEYS: Keys = Keys {
    epoch_duration: "epoch_duration",
    vp_whitelist: "vp_whitelist",
};

const ALL_KEYS: &[&str] = {
    // if something is missing, we get compiler error here
    let Keys {
        epoch_duration,
        vp_whitelist,
    } = KEYS;
    &[epoch_duration, vp_whitelist]
};
```

Reply:
I found a neat crate that creates an array with all enum variants https://crates.io/crates/enum_array

I'll push a commit using this crate in a bit